### PR TITLE
Allow overriding of the url.

### PR DESCRIPTION
**Pull request by LionOps**

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: src/index.js [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
 
-  _makeConfig: function(url, opts) {
-    const urlObj = { url };
-    return assign({}, requiredDefaults, urlObj, this.defaults, this._auth,
-      opts);
+  _makeConfig: function(endpoint, opts) {
+    const conf = assign({}, requiredDefaults, this.defaults, this._auth, opts);
+    const urlObj = url.parse(conf.url);
+
+    urlObj.pathname = path.join(urlObj.pathname, endpoint, '/');
```
Comment (beck) [inline src/index.js:22]: is the trailing slash here required?

File: src/index.js [hunk 2]
```diff
@@ -24,14 +32,12 @@ const pixabayjs = {
   },
 
   imageResultList: function(search, opts, onSuccess, onFailure) {
-    const url = 'https://pixabay.com/api';
-    const conf = this._makeConfig(url, opts);
+    const conf = this._makeConfig('', opts);
```
Comment (beck) [inline src/index.js:35]: since `_makeConfig` is getting a make-over, would it be cleaner to add `endpoint` as an option?

**Review by beck — APPROVED** [review]
The above is minor critique / small potatoes.

File: src/index.js (hunk 1 above)
Comment (LionOps) [reply src/index.js:22]: It's not required in the sense that the url will still resolve correctly, but it is better to include the trailing slash.

https://cdivilly.wordpress.com/2014/03/11/why-trailing-slashes-on-uris-are-important/

File: src/index.js (hunk 2 above)
Comment (LionOps) [reply src/index.js:35]: Options come from the consumer, and it would defeat the purpose of having `imageResultList` and `videoResultList`

Comment (beck) [reply src/index.js:35]: That almost sounds like a simplification/improvement vs a downside.

**Pull request merged**